### PR TITLE
Mms 770

### PR DESCRIPTION
**Pull request by EDKarlsson**

### CODE REVIEW

File: mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/ElasticHelper.java [hunk 1]
```diff
@@ -2,13 +2,10 @@
...
 import java.net.UnknownHostException;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
 import com.google.gson.JsonElement;
+import io.searchbox.core.*;
```
Comment (HuiJun) [inline mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/ElasticHelper.java]: Explicitly import

File: mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/ElasticHelper.java [hunk 2]
```diff
@@ -464,4 +455,62 @@ private static JSONObject removeWrapper(JSONObject jsonObject) {
...
+     * @param id
+     * @return JSON Response
+     */
+    public JSONObject deleteElasticElements(String field, String id){
+        JestResult result = null;
+        JSONObject query = new JSONObject();
+        query.put("query", new JSONObject().put("term", new JSONObject().put(field, id)));
+
+        // Verbose statement to make sure it uses the correct delete by query class from searchbox.
+        DeleteByQuery deleteByQuery = new io.searchbox.core.DeleteByQuery.Builder(query.toString()).addIndex(elementIndex).build();
```
Comment (HuiJun) [inline mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/ElasticHelper.java]: Clean up

File: mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresHelper.java [hunk 3]
```diff
@@ -2,11 +2,7 @@
 
 import java.beans.PropertyVetoException;
 import java.io.IOException;
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Savepoint;
+import java.sql.*;
```
Comment (HuiJun) [inline mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresHelper.java]: Explicit import

File: mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresHelper.java [hunk 4]
```diff
@@ -2100,4 +2096,46 @@ public boolean refExists(String refId) {
...
+        closeConfig();
+    }
+
+    /**
+     * Deletes the project from the project table based on the projectId provided.
+     * @param projectId
+     */
+    public void deleteProjectFromProjectsTable(String projectId){
+        String query = "DELETE FROM projects WHERE projectid = \'" + projectId +"\'";
+        System.out.println("Delete ProjectId Query "  + query);
```
Comment (HuiJun) [inline mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresHelper.java]: Use prepared statement here

File: mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresPool.java [hunk 5]
```diff
@@ -69,4 +69,18 @@ public static Connection getStandaloneConnection(String host, String name) throw
...
+
+    /**
+     * Given a host and name this will remove the connection source from the PostgresPool
+     * @param host
+     * @param name
+     */
+    public static void removeConnection(String host, String name) {
+        String connectString = host + name;
+
+        System.out.println("Removing connection " + connectString);
```
Comment (HuiJun) [inline mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresPool.java]: Remove

File: mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresHelper.java (hunk 4 above)
Comment (HuiJun) [reply mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresHelper.java]: Also remove println

**Review by HuiJun — CHANGES_REQUESTED** [review]

File: mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresHelper.java (hunk 4 above)
Comment (EDKarlsson) [reply mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresHelper.java]: I tried making this a prepared statement but it throws a null pointer exception on the `PreparedStatment query =...` line.
```java
PreparedStatement query = this.conn.prepareStatement("DELETE FROM projects WHERE projectid = ?");
query.setString(1, projectId);
query.executeUpdate();
```
Exception
```
2017-06-29 08:47:58,221  ERROR [view_repo.webscripts.ProjectDelete] [http-bio-8080-exec-1] java.lang.NullPointerException
	at gov.nasa.jpl.view_repo.db.PostgresHelper.deleteProjectFromProjectsTable(PostgresHelper.java:2137)
	at gov.nasa.jpl.view_repo.webscripts.ProjectDelete.deleteProjectFromProjectsTable(ProjectDelete.java:177)
	at gov.nasa.jpl.view_repo.webscripts.ProjectDelete.executeImplImpl(ProjectDelete.java:72)
	at gov.nasa.jpl.view_repo.webscripts.ProjectDelete.executeImpl(ProjectDelete.java:47)
	at gov.nasa.jpl.view_repo.webscripts.DeclarativeJavaWebScript.execute(DeclarativeJavaWebScript.java:103)
   ...
```

File: mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/ElasticHelper.java (hunk 1 above)
Comment (EDKarlsson) [reply mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/ElasticHelper.java]: Fixed

**Review by HuiJun — APPROVED** [review]

**Pull request merged**